docs/source/Coding/Setting-up-PyCharm.md

```diff
@@ -1,57 +1,104 @@
-1. Open your project in PyCharm
-1. In the PyCharm menu, select `Run > Attach to Local Process...`
-1. From the list, pick the `twistd` process with the `server.py` parameter (Example: `twistd.exe --nodaemon --logfile=\<mygame\>\server\logs\server.log --python=\<evennia repo\>\evennia\server\server.py`)
+### Attaching to the process
+1. Launch Evennia in the pycharm terminal
+2. Attempt to start it twice, this will give you the process ID of the server
+3. In the PyCharm menu, select `Run > Attach to Process...`
+4. From the list, pick the corresponding process id, it should be the `twistd` process with the `server.py` parameter (Example: `twistd.exe --nodaemon --logfile=\<mygame\>\server\logs\server.log --python=\<evennia repo\>\evennia\server\server.py`)
 
-Of course you can attach to the `portal` process as well.  If you want to debug the Evennia launcher
+You can attach to the `portal` process as well, if you want to debug the Evennia launcher
 or runner for some reason (or just learn how they work!), see Run Configuration below.
 
 > NOTE: Whenever you reload Evennia, the old Server process will die and a new one start. So when you restart you have to detach from the old and then reattach to the new process that was created.
 
-> To make the process less tedious you can apply a filter in settings to show only the server.py process in the list. To do that navigate to: `Settings/Preferences | Build, Execution, Deployment | Python Debugger` and then in `Attach to process` field put in: `twistd.exe" --nodaemon`. This is an example for windows, I don't have a working mac/linux box.
 
-![Example process filter configuration](https://i.imgur.com/vkSheR8.png)
-
-## Run Evennia from inside PyCharm
+### Run Evennia with a Run/Debug Configuration
 
 This configuration allows you to launch Evennia from inside PyCharm. Besides convenience, it also allows suspending and debugging the evennia_launcher or evennia_runner at points earlier than you could by running them externally and attaching. In fact by the time the server and/or portal are running the launcher will have exited already.
 
+#### On Windows
 1. Go to `Run > Edit Configutations...`
-1. Click the plus-symbol to add a new configuration and choose Python
-1. Add the script: `\<yourrepo\>\evenv\Scripts\evennia_launcher.py` (substitute your virtualenv if it's not named `evenv`)
-1. Set script parameters to: `start -l` (-l enables console logging)
-1. Ensure the chosen interpreter is from your virtualenv
-1. Set Working directory to your `mygame` folder (not evenv nor evennia)
-1. You can refer to the PyCharm documentation for general info, but you'll want to set at least a config name (like "MyMUD start" or similar).
+2. Click the plus-symbol to add a new configuration and choose Python
+3. Add the script: `\<yourprojectfolder>\.venv\Scripts\evennia_launcher.py` (substitute your virtualenv if it's not named `evenv`)
+4. Set script parameters to: `start -l` (-l enables console logging)
```

Comment:
Out of curiosity, why is the daemon mode used and not the interactive mode (`evennia istart`)? Not saying this solution is wrong, but I'd have thought it would be better to let PyCharm have full control over the Evennia run process in foreground mode.

Reply:
Oh, I haven't touched this part, just moved it, since I'm on linux now.
Maybe I should try to make a pass with windows.